src/main/java/io/appform/dropwizard/actors/actor/UnmanagedBaseActor.java

```diff
@@ -198,7 +198,10 @@ public void handleDelivery(String consumerTag, Envelope envelope,
                 }
             } catch (Throwable t) {
                 log.error("Error processing message...", t);
-                if (errorCheckFunction.apply(t)) {
+                if(retryStrategy.postRetryAckHandling()) {
```

Comment:
Make this handling an algorithm class. To be called _after_ the exception check. Not before.